src/engine/effects/engineeffectsdelay.h

```diff
@@ -39,6 +44,17 @@ class EngineEffectsDelay final : public EngineObject {
         // EngineEffectsDelay structure works with delay samples, so the value
         // is recalculated for the EngineEffectsDelay usage.
         m_currentDelaySamples = delayFrames * mixxx::kEngineChannelCount;
+
+        VERIFY_OR_DEBUG_ASSERT(m_currentDelaySamples >= 0) {
+            m_currentDelaySamples = 0;
+        }
+
+        // When mixxx will support other channel count variants than stereo,
+        // the kMaxDelay has to be divisible by the number of channels.
+        // Otherwise, channels may be swapped.
+        VERIFY_OR_DEBUG_ASSERT(m_currentDelaySamples <= kMaxDelay - mixxx::kEngineChannelCount) {
+            m_currentDelaySamples = kMaxDelay - mixxx::kEngineChannelCount;
+        }
```

Comment:
You're subtracting here, isn't that a bug? Also I'd just add another constexpr variable `kMaxDelayFrames` instead.

Reply:
****

> You're subtracting here, isn't that a bug? Also I'd just add another constexpr variable kMaxDelayFrames instead.

The inspiration was by [EngineDelay](https://github.com/mixxxdj/mixxx/blob/0d1dc9fa448181db8714425b424a754227bc0e1b/src/engine/enginedelay.cpp#L41). The reason of subtraction is, that last valid delay value is `kMaxDelay - 1`. However, clamping delay to this value will cause a switch between channels (for example stereo channels will be swapped). However, with your proposal of `kMaxDelayFrames`, the subtraction wouldn't be needed and only frames value will be compared.